ndcube/utils/wcs.py

```diff
@@ -121,100 +123,194 @@ def _augment(cls, header, naxis):
+    Returns the new sliced wcs, changed missing axes, and coordinates of dropped axes.
 
     Paramters
     ---------
     wcs: `astropy.wcs.WCS` or `ndcube.utils.wcs.WCS`
         WCS object to be sliced.
 
     missing_axes: `list` of `bool`
-        Indicates which axes of the WCS are "missing", i.e. do not correspond to a data axis.
+        Indicates which axes of the WCS are "missing" i.e. do not correspond to a data axis.
+        Must be supplied in WCS order.
 
     item: `int`, `slice` or `tuple` of `int` and/or `slice`.
-        Slicing item.  Note that unlike in other places in this package, the item has the
-        same axis ordering as the WCS object, i.e. the reverse of the data order.
+        Slicing item. Must be supplied in numpy order and not include entries for missing axes.
+        See Notes for further explanation.
 
     Returns
     -------
     new_wcs: `astropy.wcs.WCS` or `ndcube.utils.wcs.WCS`
         Sliced WCS object.
 
-    missing_axes: `list` of `bool`
-        Altered missing axis list.  Note the ordering has been reversed to reflect the data
-        (numpy) axis ordering convention.
+    new_missing_axes: `list` of `bool`
+        Altered missing axis list in WCS order.
+
+    dropped_coords:
+        Coordinates which have been dropped in the slicing process is collected
+        in a dictionary called dropped_coords.
+
+    Notes
+    -----
+    A clarifying example of axis ordering of wcs, missing_axes and item.
+
+    Let's say we have a wcs object with four axes (2, 1, 5, 4) and a
+    missing_axes list of [False, True, False, False].
+    This means that the 1th axis of the wcs is "missing".
+    Note that the ordering of these axes are in WCS order, i.e. the reverse of
+    numpy order which is the order the data axes are given.
+    So the data shape must be (4, 5, 2), given that we have one "missing" axis.
+    The item object must be in numpy order and mustn't account for missing axes.
+    So a valid item object would be (slice(2, 4), slice(0, 1), slice(None)),
+    where the 0th entry gives the slice to be applied the WCS axis of length 4,
+    the middle slice is applied to WCS axis of length 5,
+    the missing axis is not represented,
+    and the last entry is applied to the WCS axis to length 2.
 
     """
-    # normal slice.
+    # item is entered in numpy order and does not include entries for missing axes.
+    # But it is easier for it to be in WCS order and include entries for missing axes
+    # for extracting real world coordinates of dropped axes.
+    # Due to the possibility of missing axes, the best thing to do
+    # is first reverse the missing_axes variable, then prep the item
+    # to account for those missing axes and any non-missing axes not to be sliced.
+    # This will make the item the same length as the number of axes in the WCS object
+    # meaning it can be safely reversed to WCS order.
+    missing_axes_numpy_order = missing_axes[::-1]
```

Comment:
In order to stop the code implicitly changing the original input ```missing_axes```, we need to create a separate copy.  Suggest the following change.
```python
new_missing_axes = deepcopy(missing_axes)
missing_axes_numpy_order = new_missing_axes[::-1]
```